core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/CleanupFreeIpaHandler.java

```diff
@@ -51,13 +53,15 @@ public void accept(Event<CleanupFreeIpaEvent> cleanupFreeIpaEvent) {
                 freeIpaCleanupService.cleanupOnScale(stack, event.getHostNames(), event.getIps());
             }
             LOGGER.debug("Cleanup finished for hosts: {} and IPs: {}", event.getHostNames(), event.getIps());
-        } catch (Exception e) {
-            LOGGER.error("FreeIPA cleanup failed for hosts {} and IPs: {}", event.getHostNames(), event.getIps(), e);
-        } finally {
             CleanupFreeIpaEvent response = new CleanupFreeIpaEvent(CLEANUP_FREEIPA_FINISHED_EVENT.event(), event.getResourceId(), event.getHostNames(),
                     event.getIps(), event.isRecover());
             Event<StackEvent> responseEvent = new Event<>(cleanupFreeIpaEvent.getHeaders(), response);
             eventBus.notify(CLEANUP_FREEIPA_FINISHED_EVENT.event(), responseEvent);
+        } catch (Exception e) {
+            LOGGER.error("FreeIPA cleanup failed for hosts {} and IPs: {}", event.getHostNames(), event.getIps(), e);
+            StackFailureEvent response = new StackFailureEvent(CLEANUP_FREEIPA_FAILED_EVENT.event(), event.getResourceId(), e);
+            Event<StackEvent> responseEvent = new Event<>(cleanupFreeIpaEvent.getHeaders(), response);
+            eventBus.notify(CLEANUP_FREEIPA_FAILED_EVENT.event(), responseEvent);
```

Comment:
Concern: on cleanup exception, the flow will always go failed, before this change we ignored the cleanup errors. What will be the side effect(s) of this?

Reply:
it should have always failed. residue from previous clusters or previous upscale/downscale can cause issues. This was a bug. It's possible we would see some failures caused by this, but it's better then seeing other issues.